homeassistant/components/growatt/sensor.py

```diff
@@ -0,0 +1,179 @@
+"""Support for Growatt Plant energy production sensors."""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+POWER_KILO_WATT = "kW"
+ENERGY_MEGA_WATT_HOUR = "MWh"
+ENERGY_GIGA_WATT_HOUR = "GWh"
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt Plant sensor."""
+    import growatt
+
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    growatt_client = growatt.GrowattApi()
+
+    if login(growatt_client, username, password):
+        sensor_today = GrowattPlantTotals(
+            hass,
+            growatt_client,
+            username,
+            password,
+            "Growatt plant today",
+            "todayEnergySum",
+        )
+        sensor_total = GrowattPlantTotals(
+            hass,
+            growatt_client,
+            username,
+            password,
+            "Growatt plant total",
+            "totalEnergySum",
+        )
+        sensor_current = GrowattPlantCurrent(
+            hass, growatt_client, username, password
+        )
+        add_entities([sensor_today, sensor_total, sensor_current])
+
+
+def login(client, username, password):
+    """Login to the growatt server."""
+    import growatt
+
+    try:
+        client.login(username, password)
+        return True
+    except growatt.LoginError as error:
+        _LOGGER.error(error)
+        return False
+
+
+class GrowattPlant(Entity):
+    """Base Growatt sensor class."""
+
+    def __init__(self, hass, client, username, password):
```

Comment:
Do not pass in `hass`.  It is available as `self.hass` after the entity is added.  Please change here and in the derived classes.